frontend/src/Pages/PostPage.tsx

```diff
@@ -54,16 +55,20 @@ export default function PostPage() {
         }
 
         let scrollToComment: HTMLDivElement | null | undefined;
+        let commentId: number | undefined;
         if (location.hash) {
-            const commentId = parseInt(location.hash.substring(1));
+            commentId = parseInt(location.hash.substring(1));
             scrollToComment = document.querySelector<HTMLDivElement>(`[data-comment-id="${commentId}"]`);
         }
         else if (unreadOnly) {
             // find first new comment
              scrollToComment = document.querySelector<HTMLDivElement>(`.isNew`);
+             commentId = scrollToComment?.dataset.commentId ? parseInt(scrollToComment.dataset.commentId) : undefined;
         }
         // do nothing if element is focused already
-        if(!scrollToComment || scrollToComment.className.indexOf(styles.focusing) >= 0) {
+        if (!scrollToComment || scrollToComment.className.indexOf(styles.focusing) >= 0 ||
+            (scrolledToComment && scrolledToComment.postId === postId && scrolledToComment.commentId === commentId)
```

Comment:
`scrolledToComment` is a non-null here alread
```suggestion
            scrolledToComment.postId === postId && scrolledToComment.commentId === commentId
```

Reply:
Oops, missed that. If you have time, please refactor it.